basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -870,6 +870,14 @@ namespace basic_autonomy
             lanelet::BasicLineString2d new_points = create_lanechange_path(lanelets_in_path[lane_change_iteration],lanelets_in_path[lane_change_iteration + 1]);
             centerline_points.insert(centerline_points.end(), new_points.begin(), new_points.end());
 
+            //Add points from following lanelet to provide sufficient distance for adding buffer
+            auto following_lanelets = wm->getMapRoutingGraph()->following(lanelets_in_path[lane_change_iteration + 1], false);
+            if(!following_lanelets.empty()){
+                //Arbitrarily choosing first following lanelet for buffer since points are only being used to fit spline
+                centerline_points.insert(centerline_points.end(), following_lanelets.front().centerline2d().basicLineString().begin(), 
+                                                                            following_lanelets[0].centerline2d().basicLineString().end());
```

Comment:
I'm a bit concerned about this call
following_lanelets.front().centerline2d().basicLineString()
When this is called is the returned vector always at the same address or is it copied out as a new vector. If the latter then these iterators (begin(), end()) might not point to the same data set. I think it would be safer to store the basicLineString first then access the iterators